truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/nodes/Node.java

```diff
@@ -524,9 +525,65 @@ public final void atomic(Runnable closure) {
     }
 
     /**
+     * Returns <code>true</code> if this node should be considered tagged by a given tag else
+     * <code>false</code>. The method is only invoked for tags which are explicitly declared as
+     * {@link com.oracle.truffle.api.instrumentation.ProvidedTags provided} by the
+     * {@link TruffleLanguage language}. If the {@link #getSourceSection() source section} of the
+     * node returns <code>null</code> then this node is considered to be not tagged by any tag.
+     * <p>
+     * Tags are used by guest languages to indicate that a {@link Node node} is a member of a
+     * certain category of nodes. For example a debugger
+     * {@link com.oracle.truffle.api.instrumentation.TruffleInstrument instrument} might require a
+     * guest language to tag all nodes as halt locations that should be considered as such.
+     * <p>
+     * The node implementor may decide how to implement tagging for nodes. The simplest way to
+     * implement tagging using Java types is by overriding the {@link #isTaggedWith(Class)} method.
+     * This example shows how to tag a node subclass and all its subclasses as expression and
+     * statement:
+     *
+     * <pre>
+     * <code>
+     * &#64;{@link Override}
+     * protected boolean isTaggedWith({@link Class} tag) {
+     *    return tag == ExpressionTag.class || tag == StatementTag.class;
+     * }
+     * </code>
+     * </pre>
+     *
+     * <p>
+     * Often it is impossible to just rely on the node's Java type to implement tagging. This
+     * example shows how to use local state to implement tagging for a node.
+     *
+     * <pre>
+     * <code>
+     * private boolean isDebuggerHalt;
```

Comment:
Can we use code snippets, please? You can copy one from: https://github.com/jtulach/truffle/commit/7b30201efb440636858daf695c03539d32706fc1#diff-390d3d3d22a24a2b4a58c1918871c59fR704